File: src/main/java/com/imagegame/service/GameSessionService.java
package com.imagegame.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.imagegame.domain.GameSession;
import com.imagegame.dto.GameSessionDTO;
import com.imagegame.dto.GameSessionSearchDTO;
import com.imagegame.dto.GameSessionPageDTO;
import com.imagegame.dto.GameSessionConvertCriteriaDTO;
import com.imagegame.service.GenericService;
import com.imagegame.dto.common.RequestDTO;
import com.imagegame.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface GameSessionService extends GenericService<GameSession, Integer> {

	List<GameSession> findAll();

	ResultDTO addGameSession(GameSessionDTO gameSessionDTO, RequestDTO requestDTO);

	ResultDTO updateGameSession(GameSessionDTO gameSessionDTO, RequestDTO requestDTO);

    Page<GameSession> getAllGameSessions(Pageable pageable);

    Page<GameSession> getAllGameSessions(Specification<GameSession> spec, Pageable pageable);

	ResponseEntity<GameSessionPageDTO> getGameSessions(GameSessionSearchDTO gameSessionSearchDTO);
	
	List<GameSessionDTO> convertGameSessionsToGameSessionDTOs(List<GameSession> gameSessions, GameSessionConvertCriteriaDTO convertCriteria);

	GameSessionDTO getGameSessionDTOById(Integer gameSessionId);







}

File: imagegame_ui/src/services/GameSessionService.js
import http from "../http-common"; 

class GameSessionService {
  getAllGameSessions(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/gameSession/gameSessions`, searchDTO);
  }

  get(gameSessionId) {
    return this.getRequest(`/gameSession/${gameSessionId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/gameSession?field=${matchData}`, null);
  }

  addGameSession(data) {
    return http.post("/gameSession/addGameSession", data);
  }

  update(data) {
  	return http.post("/gameSession/updateGameSession", data);
  }
  
  uploadImage(data,gameSessionId) {
  	return http.postForm("/gameSession/uploadImage/"+gameSessionId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new GameSessionService();
